snakemake/common/prefix_lookup.py

```diff
@@ -25,27 +25,26 @@ def match_iter(self, query: str) -> Generator[V, None, None]:
         """
         stop_idx = len(self._entries)
         while stop_idx:
-            stop_idx = bisect.bisect_right( self._entries,
-                                            query,
-                                            hi=stop_idx,
-                                            key=lambda x: x[0])
+            stop_idx = bisect.bisect_right(
+                self._entries, query, hi=stop_idx, key=lambda x: x[0]
+            )
 
-            k, entries = self._entries[stop_idx-1]
+            k, entries = self._entries[stop_idx - 1]
             if query.startswith(k):
                 for entry in entries:
                     yield entry
 
-            if not query:
-                # Prevent infinite loop if '' is first string in list
+            if not query or not k:
```

Comment:
@tbooth I've added this condition here, do you agree with that?

Reply:
I saw that too, but I think we can safely ignore code rabbit suggestions, if they aren't helpful like in this case. They keep coming in as we push commits.